### ISSUE TITLE
Use new urdf SharedPtr typedefs

### ISSUE THREAD

**Issue by nxdefiant** [moveit/srdfdom#17]
Urdf has moved to a new SharedPtr type which results in a compile error:

```
/opt/ros/ros_catkin_ws/src/srdfdom/src/model.cpp: In member function 'void srdf::Model::loadGroups(const urdf::ModelInterface&, TiXmlElement*)':
/opt/ros/ros_catkin_ws/src/srdfdom/src/model.cpp:182:43: error: variable 'boost::shared_ptr<const urdf::Link> l' has initializer but incomplete type
       boost::shared_ptr<const urdf::Link> l = urdf_model.getLink(tip_str);
```

Replacing 
`boost::shared_ptr<const urdf::Link> l`
with
`urdf::LinkConstSharedPtr l`

fixes this error.

**Comment by rhaschke** [comment]
Addressed in #18. Note, that urdfdom 1.0 (that has the transition to std::shared_ptr) is not released yet in any Ubuntu release. Xenial/Kinetic still comes with version 0.4.1.

**Comment by nxdefiant** [comment]
But Debian testing/unstable already has it. #18 compiles, thanks.
